redbox/storage/elasticsearch.py

```diff
@@ -141,3 +142,44 @@ def get_file_chunks(self, parent_file_uuid: UUID) -> list[Chunk]:
             )
         ]
         return res
+
+    def _count_chunks(self, parent_file_uuid: UUID) -> int:
+        target_index = f"{self.root_index}-chunk"
+        res = self.es_client.count(
+            index=target_index,
+            body={"query": {"match": {"parent_file_uuid": str(parent_file_uuid)}}},
+        )
+        return res["count"]
+
+    def _count_embedded_chunks(self, parent_file_uuid: UUID) -> int:
+        target_index = f"{self.root_index}-chunk"
+        res = self.es_client.count(
+            index=target_index,
+            body={
+                "query": {
+                    "bool": {
+                        "must": [
+                            {"match": {"parent_file_uuid": str(parent_file_uuid)}},
+                            {"exists": {"field": "embedding"}},
+                        ]
+                    }
+                }
+            },
+        )
+        return res["count"]
+
+    def get_file_status(self, file_uuid: UUID):
+        file = self.read_item(file_uuid, model_type="File")
+        if file.processing_status == "embedding":
+            chunk_count = self._count_chunks(file_uuid)
+            embedded_chunk_count = self._count_embedded_chunks(file_uuid)
+
+            if chunk_count == embedded_chunk_count:
+                file.processing_status = ProcessingStatusEnum.complete
+                self.update_item(file_uuid, file)
```

Comment:
im not sure about this, because this means that the `File.status` gets updates if , and only if, someone/thing calls `get_file_status`. What if this is never called, and a downstream process relies upon the `File.status` being accurate?

some ideas:
1. we could drop the `File.status` field altogether and could ensure that the only way to to get the status is to call `get_file_status` either directly or via the endpoint
2. we could have the `embedder`/`ingester` update the `File.status` field

either way I dont think that asking for the file status should change it.

Reply:
another random thought... what if the FileStatus model looked like this:

```python

class ChunkStatus(BaseModel):
    uuid: UUID
    embedded: bool
    
class FileStatus(BaseModel):
    uuid: UUID
    chunks: Optional[list[ChunkStatus]]
```

the rationale being:
1. if the file exists then it is uploaded
2. lets do away with the idea of uploading a file that we will never be chunked/embedded, we have no use case for this and even if we do it doesnt need to be handled by the backend?
3. if `File.chunks` is None then this implies that the file is still chunking
4. the backend should return the maximum information, like how many chunks there are and how many have been embedded but the minimum English. Its up to the various front ends to best interpret this for their users

Reply:
I like this mental model, much cleaner. Lets have a crack at it on Tuesday